cou/steps/analyze.py

```diff
@@ -50,29 +55,33 @@ async def create(cls) -> Analysis:
         return Analysis(apps=apps)
 
     @classmethod
-    async def _populate(cls) -> List[Application]:
+    async def _populate(cls) -> list[OpenStackApplication]:
         """Analyze the applications in the model.
 
         :return: Application objects with their respective information.
-        :rtype: List[Application]
+        :rtype: List[OpenStackApplication]
         """
         juju_status = await async_get_status()
         model_name = juju_status.model.name
         apps = {
-            Application(
+            AppFactory.create(
                 name=app,
                 status=app_status,
                 config=await async_get_application_config(app),
                 model_name=model_name,
+                charm=await extract_charm_name(app),
             )
             for app, app_status in juju_status.applications.items()
         }
-        upgradeable_apps = {app for app in apps if app.charm in UPGRADE_ORDER}
+        # remove non-supported charms that return None on AppFactory.create
+        apps.remove(None)
+        upgradeable_apps = {app for app in apps if app and app.charm in UPGRADE_ORDER}
         unknown_apps = apps - upgradeable_apps
         upgradeable_apps_sorted = sorted(
             upgradeable_apps, key=lambda app: UPGRADE_ORDER.index(app.charm)
         )
-        return upgradeable_apps_sorted + list(unknown_apps)
+        # mypy complains that unknow_apps can have None, but we already removed None from apps
+        return upgradeable_apps_sorted + list(unknown_apps)  # type: ignore
```

Comment:
`plan.py` will try to generate plans for unknown_apps (where unknown = apps I can lookup in my app/codename table, but are not in the `UPGRADE_ORDER` - maybe we can find a better name). Is this ok?

Reply:
I've changed to:

- `apps_to_upgrade_in_order` (apps that we know that have a order to upgrade)
- `sorted_apps_to_upgrade_in_order` (apps that we know that have a order to upgrade sorted in the right sequence)
- `disordered_upgrade_apps` (apps that can upgrade, but we don't care about the order)
- `disordered_upgrade_apps_sorted_by_name` (apps that can upgrade, but we don't care about the order and are sorted by charm name)